controllers/dataingestendpointsecret/secret_test.go

```diff
@@ -0,0 +1,276 @@
+
+		upd, err := endpointSecretGenerator.GenerateForDynakube(context.TODO(), instance)
+		assert.NoError(t, err)
+		assert.Equal(t, true, upd)
+
+		updateSecret(t, fakeClient)
+
+		upd, err = endpointSecretGenerator.GenerateForDynakube(context.TODO(), instance)
+		assert.NoError(t, err)
+		assert.Equal(t, true, upd)
+
+		checkSecretExists(t, fakeClient, webhook.SecretEndpointName, testNamespace1, testUpdatedTokenDataIngestSecret)
+
+		checkSecretExists(t, fakeClient, webhook.SecretEndpointName, testNamespace2, testUpdatedTokenDataIngestSecret)
+
+		checkSecretNotExists(t, fakeClient, webhook.SecretEndpointName, testNamespaceDynatrace)
+	})
+	t.Run(`data-ingest endpoint secret created in all namespaces and apiUrl updated`, func(t *testing.T) {
+		instance := buildDynakube()
+		fakeClient := buildClient(instance)
+
+		endpointSecretGenerator := NewEndpointGenerator(fakeClient, fakeClient, testNamespaceDynatrace, log)
+
+		upd, err := endpointSecretGenerator.GenerateForDynakube(context.TODO(), instance)
+		assert.NoError(t, err)
+		assert.Equal(t, true, upd)
+
+		newInstance := updatedDynakube()
+
+		upd, err = endpointSecretGenerator.GenerateForDynakube(context.TODO(), newInstance)
+		assert.NoError(t, err)
+		assert.Equal(t, true, upd)
+
+		checkSecretExists(t, fakeClient, webhook.SecretEndpointName, testNamespace1, testUpdatedApiUrlDataIngestSecret)
+
+		checkSecretExists(t, fakeClient, webhook.SecretEndpointName, testNamespace2, testUpdatedApiUrlDataIngestSecret)
+
+		checkSecretNotExists(t, fakeClient, webhook.SecretEndpointName, testNamespaceDynatrace)
+	})
+}
+
+func checkSecretExists(t *testing.T, fakeClient client.Client, secretName string, namespace string, data string) {
+	var testSecret corev1.Secret
+	err := fakeClient.Get(context.TODO(), client.ObjectKey{Name: secretName, Namespace: namespace}, &testSecret)
+	assert.NoError(t, err)
+	assert.NotNil(t, testSecret.Data)
+	assert.NotEmpty(t, testSecret.Data)
+	assert.Contains(t, testSecret.Data, "endpoint.properties")
+	assert.Equal(t, data, string(testSecret.Data["endpoint.properties"]))
+}
+
+func checkSecretNotExists(t *testing.T, fakeClient client.Client, secretName string, namespace string) {
+	var testSecret corev1.Secret
+	err := fakeClient.Get(context.TODO(), client.ObjectKey{Name: secretName, Namespace: namespace}, &testSecret)
+	assert.Error(t, err)
+	assert.Nil(t, testSecret.Data)
+}
+
+func updateSecret(t *testing.T, fakeClient client.Client) {
+	updatedSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testDynakubeName,
+			Namespace: testNamespaceDynatrace,
+		},
+		Data: map[string][]byte{
+			"apiToken":        []byte(testAPIToken),
+			"paasToken":       []byte(testPaasToken),
+			"dataIngestToken": []byte(testUpdatedDataIngestToken),
+		},
+	}
+
+	err := fakeClient.Update(context.TODO(), updatedSecret)
+	assert.NoError(t, err)
+}
+
+func updatedDynakube() *dynatracev1beta1.DynaKube {
```

Comment:
should have `test` (something like `updatedTestDynakube` or something) in the name because it looks too much like actual code and not testing code 
(this is true for most of the functions in this file)

Reply:
functions renamed